### DISCUSSION TITLE
Programatically select specific tab?

### DISCUSSION THREAD

**Discussion by dprothero** [code-hike/codehike#412] [Q&A]
We have a CH.Code with multiple code samples to be rendered as tabs. We are using each tab for a particular programming language:
![image](https://github.com/code-hike/codehike/assets/1897006/8c815674-8eeb-46b8-a123-0e2544df3c11)

What I'd like to do is pre-select a tab based on a previously saved programming language preference. I've tried wrapping the Code component to experiment like so:

```jsx
const TwilioCode = (props: CodeProps) => {
  const { codeLanguage, setCodeLanguage } = useContext(DocsContext);
  props.northPanel.active = codeLanguage;

  const handleClick = () => {
    console.log('boop');
    setCodeLanguage('Java');
    props.northPanel.active = 'Java';
  };
  return (
    <>
      <button type="button" onClick={handleClick}>
        Boop
      </button>
      <CH.Code {...props} />
    </>
  );
};
```

It renders properly, but there doesn't seem to be a way to make the selected tab change.

In addition, I will need to be able to save their preference if they change tabs.

Any ideas?

**Comment by pomber** [comment]
There's no good way of doing that currently. But **I'm working on v1.0 and the idea is to include a built-in switcher for that use case.** 

Also, if you have more feature requests this is a great time to bring them up :)

**Reply by stern-shawn** [reply]
Looking forward to it!

We're currently hacking around it with `useEffect`, watching for click events on the tabs based on their `title` and `data-ch-tab` attributes, and doing synthetic clicks on other matching tabs in response. It _works_, but it'd feel much nicer to update a prop instead 😄

**Reply by pomber** [reply]
I was going to suggest something like that but discarded it as too hacky 😆

**Reply by stern-shawn** [reply]
Haha, extremely hacky and not terribly performant with many samples on a page... 😅 
We'd be happy to collaborate on a change depending on where the work for v1 is at currently 😃

**Reply by pomber** [reply]
Here is one example of the switcher with v1 (work in progress): https://v1-outpost.vercel.app/api-reference.  
Code is here: [code-switcher.tsx](https://github.com/code-hike/v1/blob/main/apps/outpost/app/api-reference/code-switcher.tsx) 

Some questions for you, is the thing you are building open source? are you using React Server Components? are you interested in early access to Code Hike v1?

**Reply by stern-shawn** [reply]
- We are still using the pages router for now; the site is a mix of SSG and ISR based on next-mdx-remote for context (so all content is purely MDX and we cannot use typical component imports). RSC may make sense in the future!
- The project currently isn't open source as it's the new docs platform for Twilio.

@dprothero this is pretty neat!
